### PR TITLE
implemented user parking setting feature

### PR DESCRIPTION
**Pull request by Faka-GLB**
Implemented a DialogFragment to set a number of parking lots, the fragment has a mvp of its own, but without a model because it was unnecessary.  This new mvp also has contracts, and a interface was declared on the DialogFragment, it is used to communicate the number of parking lots with the activity mvp.

image

fragment_dialog

### CODE REVIEW

File: app/src/main/java/com/onboarding/parkingsystemjava/fragment/SetParkingLotsDialog.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+import android.app.DialogFragment;
+import android.content.Context;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.EditText;
+
```
Comment (lucianoeli) [inline app/src/main/java/com/onboarding/parkingsystemjava/fragment/SetParkingLotsDialog.java]: remove empty lines

File: app/build.gradle [hunk 2]
```diff
@@ -12,7 +12,7 @@ android {
 
     defaultConfig {
         applicationId "com.onboarding.parkingsystemjava"
-        minSdkVersion 23
+        minSdkVersion 25
```
Comment (nicolbidone) [inline app/build.gradle]: why do you change the min sdk?

File: app/src/main/java/com/onboarding/parkingsystemjava/activity/MainActivity.java [hunk 3]
```diff
@@ -1,18 +1,21 @@
 package com.onboarding.parkingsystemjava.activity;
 
 import android.os.Bundle;
+import android.util.Log;
 
 import androidx.appcompat.app.AppCompatActivity;
```
Comment (nicolbidone) [inline app/src/main/java/com/onboarding/parkingsystemjava/activity/MainActivity.java]: please don't use line breaks between imports

File: app/src/main/java/com/onboarding/parkingsystemjava/fragment/SetParkingLotsDialog.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        cancelButton.setOnClickListener(view -> presenter.OnCancelButtonPress());
+    }
+
+    @Override
+    public void onAttach(Context context) {
+        super.onAttach(context);
+        try {
+            inputListener = (OnInputListener) getActivity();
+        } catch (ClassCastException e) {
+            Log.e(TAG, "ClassCastException: " + e.getMessage());
```
Comment (nicolbidone) [inline app/src/main/java/com/onboarding/parkingsystemjava/fragment/SetParkingLotsDialog.java]: here you can use String.format
https://www.baeldung.com/java-strings-concatenation#2stringformat
and please avoid harcode

File: app/src/main/java/com/onboarding/parkingsystemjava/fragment/SetParkingLotsDialog.java [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+import android.widget.EditText;
+
+import androidx.annotation.Nullable;
+
+import com.onboarding.parkingsystemjava.R;
+import com.onboarding.parkingsystemjava.mvp.contract.SetParkingContract;
+import com.onboarding.parkingsystemjava.mvp.presenter.SetParkingPresenter;
+import com.onboarding.parkingsystemjava.mvp.view.SetParkingView;
+
+public class SetParkingLotsDialog extends DialogFragment {
```
Comment (nicolbidone) [inline app/src/main/java/com/onboarding/parkingsystemjava/fragment/SetParkingLotsDialog.java]: please do not use set or get on a class name

File: app/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -3,4 +3,9 @@
     <string name="main_activity_text_view_title">EstacionAR</string>
     <string name="main_activity_button_set_parking">Establecer estacionamientos</string>
     <string name="main_activity_toast_set_parking_confirmed">Se establecieron %1$d estacionamientos.</string>
+    <string name="dialog_fragment_title">Establecer estacionamientos</string>
+    <string name="dialog_fragment_button_ok">OK</string>
+    <string name="dialog_fragment_button_cancel">Cancelar</string>
+    <string name="dialog_fragment_textview_hint">Ingrese cantidad de espacios…</string>
+    <string name="dialog_fragment_toast_empty_input">Por favor ingrese un numero.</string>
```
Comment (nicolbidone) [inline app/src/main/res/values/strings.xml]: what will happen if you have another dialog fragment? please use more descriptive names

File: app/src/main/java/com/onboarding/parkingsystemjava/mvp/contract/MainActivityContract.java [hunk 7]
```diff
@@ -4,14 +4,19 @@
...
-        void setParkingLots();
+        void setParkingLots(int setLots);
+
         int getParkingLots();
     }
 
     interface MainActivityView {
         void showParkingLotsAvailable(int parkingLots);
+
+        void showSetParkingLotsDialogFragment();
```
Comment (nicolbidone) [inline app/src/main/java/com/onboarding/parkingsystemjava/mvp/contract/MainActivityContract.java]: this show or this set?

File: app/src/main/java/com/onboarding/parkingsystemjava/mvp/view/ParkingView.java [hunk 8]
```diff
@@ -2,16 +2,23 @@
...
+import com.onboarding.parkingsystemjava.fragment.SetParkingLotsDialog;
 import com.onboarding.parkingsystemjava.mvp.contract.MainActivityContract;
 import com.onboarding.parkingsystemjava.mvp.view.base.ActivityView;
 
-public class ParkingView extends ActivityView implements MainActivityContract.MainActivityView {
+import java.util.Objects;
+
+public class
+
+ParkingView extends ActivityView implements MainActivityContract.MainActivityView {
```
Comment (nicolbidone) [inline app/src/main/java/com/onboarding/parkingsystemjava/mvp/view/ParkingView.java]: why do you set a line break here?

File: app/src/main/java/com/onboarding/parkingsystemjava/mvp/view/ParkingView.java [hunk 9]
```diff
@@ -25,4 +32,11 @@ public void showParkingLotsAvailable(int parkingLots) {
             Toast.makeText(context, context.getString(R.string.main_activity_toast_set_parking_confirmed, parkingLots), Toast.LENGTH_LONG).show();
         }
     }
+
+    @Override
+    public void showSetParkingLotsDialogFragment() {
+        Log.d(TAG, "Showing set parking lots dialog");
```
Comment (nicolbidone) [inline app/src/main/java/com/onboarding/parkingsystemjava/mvp/view/ParkingView.java]: please avoid hardcode

File: app/src/main/java/com/onboarding/parkingsystemjava/mvp/view/SetParkingView.java [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+public class SetParkingView implements SetParkingContract.SetParkingView {
+    private WeakReference<SetParkingLotsDialog> fragmentRef;
+
+    public SetParkingView(SetParkingLotsDialog fragment) {
+        this.fragmentRef = new WeakReference<>(fragment);
+    }
+
+    @Override
+    public String getLots() {
+        TextView t = fragmentRef.get().getView().findViewById(R.id.edit_text_dialog_fragment_parking_input);
```
Comment (nicolbidone) [inline app/src/main/java/com/onboarding/parkingsystemjava/mvp/view/SetParkingView.java]: please avoid using one letter var names

**Review by nicolbidone — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/onboarding/parkingsystemjava/fragment/SetParkingLotsDialog.java (hunk 5 above)
Comment (SartiMau) [reply app/src/main/java/com/onboarding/parkingsystemjava/fragment/SetParkingLotsDialog.java]: you can change it by a ConfigureParkingSize for example ;)

File: app/src/main/java/com/onboarding/parkingsystemjava/fragment/SetParkingLotsDialog.java [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+    public void onStart() {
+        super.onStart();
+        this.presenter = new SetParkingPresenter(this.inputListener, new SetParkingView(this));
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.set_parking_dialog_fragment_main_activity, container, false);
+        input = view.findViewById(R.id.edit_text_dialog_fragment_parking_input);
```
Comment (SartiMau) [inline app/src/main/java/com/onboarding/parkingsystemjava/fragment/SetParkingLotsDialog.java]: do not use findViewById pls, use viewBinding

File: app/src/main/java/com/onboarding/parkingsystemjava/fragment/SetParkingLotsDialog.java [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+
+import com.onboarding.parkingsystemjava.R;
+import com.onboarding.parkingsystemjava.mvp.contract.SetParkingContract;
+import com.onboarding.parkingsystemjava.mvp.presenter.SetParkingPresenter;
+import com.onboarding.parkingsystemjava.mvp.view.SetParkingView;
+
+public class SetParkingLotsDialog extends DialogFragment {
+
+    private SetParkingContract.SetParkingPresenter presenter;
+    private EditText input;
```
Comment (SartiMau) [inline app/src/main/java/com/onboarding/parkingsystemjava/fragment/SetParkingLotsDialog.java]: when you implement the viewBinding, you can delete this fields

File: app/src/main/java/com/onboarding/parkingsystemjava/mvp/contract/MainActivityContract.java [hunk 13]
```diff
@@ -4,14 +4,19 @@
 
     interface MainActivityPresenter {
         void onSetParkingButtonPressed();
+
```
Comment (SartiMau) [inline app/src/main/java/com/onboarding/parkingsystemjava/mvp/contract/MainActivityContract.java]: remove these extra spaces between methods definitions

File: app/src/main/java/com/onboarding/parkingsystemjava/mvp/contract/SetParkingContract.java [hunk 14]
```diff
@@ -0,0 +1,19 @@
+package com.onboarding.parkingsystemjava.mvp.contract;
+
+public interface SetParkingContract {
+
+    public interface SetParkingPresenter {
+        void OnOkButtonPress();
+
```
Comment (SartiMau) [inline app/src/main/java/com/onboarding/parkingsystemjava/mvp/contract/SetParkingContract.java]: same here

File: app/src/main/java/com/onboarding/parkingsystemjava/mvp/presenter/SetParkingPresenter.java [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+import com.onboarding.parkingsystemjava.fragment.SetParkingLotsDialog;
+import com.onboarding.parkingsystemjava.mvp.contract.SetParkingContract;
+
+public class SetParkingPresenter implements SetParkingContract.SetParkingPresenter {
+    private SetParkingContract.SetParkingView view;
+    private SetParkingLotsDialog.OnInputListener inputListener;
+    private static final String TAG = "SetParkingPresenter";
+
+    public SetParkingPresenter(SetParkingLotsDialog.OnInputListener inputListener, SetParkingContract.SetParkingView v) {
+        this.view = v;
```
Comment (SartiMau) [inline app/src/main/java/com/onboarding/parkingsystemjava/mvp/presenter/SetParkingPresenter.java]: do not use v as name, put a more representative name pls

File: app/src/main/java/com/onboarding/parkingsystemjava/mvp/presenter/SetParkingPresenter.java [hunk 16]
```diff
@@ -0,0 +1,35 @@
...
+        this.inputListener = inputListener;
+    }
+
+    @Override
+    public void OnOkButtonPress() {
+        String lots = view.getLots();
+
+        if (!lots.equals("")) {
```
Comment (SartiMau) [inline app/src/main/java/com/onboarding/parkingsystemjava/mvp/presenter/SetParkingPresenter.java]: use an EMPTY_STRING constant, do not hardcode pls.. you can create a ConstantUtils file for this kind of things

File: app/src/main/java/com/onboarding/parkingsystemjava/mvp/view/SetParkingView.java [hunk 17]
```diff
@@ -0,0 +1,34 @@
...
+import com.onboarding.parkingsystemjava.R;
+import com.onboarding.parkingsystemjava.fragment.SetParkingLotsDialog;
+import com.onboarding.parkingsystemjava.mvp.contract.SetParkingContract;
+
+import java.lang.ref.WeakReference;
+
+public class SetParkingView implements SetParkingContract.SetParkingView {
+    private WeakReference<SetParkingLotsDialog> fragmentRef;
+
+    public SetParkingView(SetParkingLotsDialog fragment) {
```
Comment (SartiMau) [inline app/src/main/java/com/onboarding/parkingsystemjava/mvp/view/SetParkingView.java]: pass here the binding as we do in the activity

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/onboarding/parkingsystemjava/fragment/SetParkingLotsDialog.java (hunk 1 above)
Comment (Faka-GLB) [reply app/src/main/java/com/onboarding/parkingsystemjava/fragment/SetParkingLotsDialog.java]: done

File: app/build.gradle (hunk 2 above)
Comment (Faka-GLB) [reply app/build.gradle]: In trying to make the DialogFragment to work, and discussing with one of my companions, one of the differences was that we used different sdk, so i tryed changing it, and forgot to change it back.

File: app/src/main/java/com/onboarding/parkingsystemjava/activity/MainActivity.java (hunk 3 above)
Comment (Faka-GLB) [reply app/src/main/java/com/onboarding/parkingsystemjava/activity/MainActivity.java]: ctrl+alt+o added those line breaks, but I'll remove them

File: app/src/main/java/com/onboarding/parkingsystemjava/fragment/SetParkingLotsDialog.java (hunk 5 above)
Comment (Faka-GLB) [reply app/src/main/java/com/onboarding/parkingsystemjava/fragment/SetParkingLotsDialog.java]: renamed

File: app/src/main/java/com/onboarding/parkingsystemjava/fragment/SetParkingLotsDialog.java (hunk 4 above)
Comment (Faka-GLB) [reply app/src/main/java/com/onboarding/parkingsystemjava/fragment/SetParkingLotsDialog.java]: done

File: app/src/main/java/com/onboarding/parkingsystemjava/mvp/view/SetParkingView.java (hunk 10 above)
Comment (lucianoeli) [reply app/src/main/java/com/onboarding/parkingsystemjava/mvp/view/SetParkingView.java]: also you should use viewBinding

File: app/src/main/res/values/strings.xml (hunk 6 above)
Comment (Faka-GLB) [reply app/src/main/res/values/strings.xml]: strings names changed

File: app/src/main/java/com/onboarding/parkingsystemjava/mvp/contract/MainActivityContract.java (hunk 7 above)
Comment (Faka-GLB) [reply app/src/main/java/com/onboarding/parkingsystemjava/mvp/contract/MainActivityContract.java]: line 18 method was for the previous pr, forgot to delete it.

File: app/src/main/java/com/onboarding/parkingsystemjava/mvp/view/ParkingView.java (hunk 8 above)
Comment (Faka-GLB) [reply app/src/main/java/com/onboarding/parkingsystemjava/mvp/view/ParkingView.java]: I really don't know, didn't noticed that, fixed

File: app/src/main/java/com/onboarding/parkingsystemjava/mvp/view/ParkingView.java (hunk 9 above)
Comment (Faka-GLB) [reply app/src/main/java/com/onboarding/parkingsystemjava/mvp/view/ParkingView.java]: done

File: app/src/main/java/com/onboarding/parkingsystemjava/mvp/view/SetParkingView.java (hunk 10 above)
Comment (Faka-GLB) [reply app/src/main/java/com/onboarding/parkingsystemjava/mvp/view/SetParkingView.java]: switched to viewBinding, removed variable t

File: app/src/main/java/com/onboarding/parkingsystemjava/fragment/SetParkingLotsDialog.java (hunk 11 above)
Comment (Faka-GLB) [reply app/src/main/java/com/onboarding/parkingsystemjava/fragment/SetParkingLotsDialog.java]: done

File: app/src/main/java/com/onboarding/parkingsystemjava/fragment/SetParkingLotsDialog.java (hunk 12 above)
Comment (Faka-GLB) [reply app/src/main/java/com/onboarding/parkingsystemjava/fragment/SetParkingLotsDialog.java]: done

File: app/src/main/java/com/onboarding/parkingsystemjava/mvp/contract/MainActivityContract.java (hunk 13 above)
Comment (Faka-GLB) [reply app/src/main/java/com/onboarding/parkingsystemjava/mvp/contract/MainActivityContract.java]: done

File: app/src/main/java/com/onboarding/parkingsystemjava/mvp/contract/SetParkingContract.java (hunk 14 above)
Comment (Faka-GLB) [reply app/src/main/java/com/onboarding/parkingsystemjava/mvp/contract/SetParkingContract.java]: done

File: app/src/main/java/com/onboarding/parkingsystemjava/mvp/presenter/SetParkingPresenter.java (hunk 15 above)
Comment (Faka-GLB) [reply app/src/main/java/com/onboarding/parkingsystemjava/mvp/presenter/SetParkingPresenter.java]: done

File: app/src/main/java/com/onboarding/parkingsystemjava/mvp/presenter/SetParkingPresenter.java (hunk 16 above)
Comment (Faka-GLB) [reply app/src/main/java/com/onboarding/parkingsystemjava/mvp/presenter/SetParkingPresenter.java]: done

File: app/src/main/java/com/onboarding/parkingsystemjava/mvp/view/SetParkingView.java (hunk 17 above)
Comment (Faka-GLB) [reply app/src/main/java/com/onboarding/parkingsystemjava/mvp/view/SetParkingView.java]: done

File: app/src/main/java/com/onboarding/parkingsystemjava/fragment/SetParkingLotsDialog.java (hunk 4 above)
Comment (nicolbidone) [reply app/src/main/java/com/onboarding/parkingsystemjava/fragment/SetParkingLotsDialog.java]: or context.getString(R.string.base_text, fill_text)

File: app/build.gradle [hunk 18]
```diff
@@ -12,7 +12,7 @@ android {
 
     defaultConfig {
         applicationId "com.onboarding.parkingsystemjava"
-        minSdkVersion 23
+        minSdkVersion 26
```
Comment (nicolbidone) [inline app/build.gradle]: undo this change

File: app/src/main/java/com/onboarding/parkingsystemjava/activity/MainActivity.java [hunk 19]
```diff
@@ -20,12 +19,17 @@ protected void onCreate(Bundle savedInstanceState) {
         binding = ActivityMainBinding.inflate(getLayoutInflater());
         setContentView(binding.getRoot());
 
-        presenter = new ParkingPresenter(new ParkingModel(), new ParkingView(this, binding));
+        presenter = new ParkingPresenter(new ParkingModel(), new ParkingView(this));
```
Comment (SartiMau) [inline app/src/main/java/com/onboarding/parkingsystemjava/activity/MainActivity.java:22]: why did you delete the binding?

File: app/src/main/java/com/onboarding/parkingsystemjava/mvp/contract/SetParkingContract.java [hunk 20]
```diff
@@ -0,0 +1,17 @@
...
+        void OnCancelButtonPress();
+    }
+
+    interface SetParkingView {
+        String getLots();
+        void closeDialog();
+        void showEmptyInputToast();
+        void showConfirmToast(int lots);
+    }
+
```
Comment (SartiMau) [inline app/src/main/java/com/onboarding/parkingsystemjava/mvp/contract/SetParkingContract.java]: remove this extra line pls

File: app/src/main/java/com/onboarding/parkingsystemjava/mvp/presenter/ConfigureParkingPresenter.java [hunk 21]
```diff
@@ -0,0 +1,33 @@
...
+
+    public ConfigureParkingPresenter(OnInputListener inputListener, SetParkingContract.SetParkingView view) {
+        this.view = view;
+        this.inputListener = inputListener;
+    }
+
+    @Override
+    public void OnOkButtonPress() {
+        String lots = view.getLots();
+        if (!lots.equals(ConstantUtils.EMPTY_STRING)) {
```
Comment (SartiMau) [inline app/src/main/java/com/onboarding/parkingsystemjava/mvp/presenter/ConfigureParkingPresenter.java]: maybe here you can use the lots.isEmpty() instead of the equals comparison

File: app/src/main/res/values/dimens.xml [hunk 22]
```diff
@@ -2,4 +2,7 @@
 <resources>
     <dimen name="button_main_inset_vertical_size_xsmall">10dp</dimen>
     <dimen name="text_size_strings_normal">20sp</dimen>
+    <dimen name="dialog_fragment_margin_size_small">20dp</dimen>
+    <dimen name="dialog_fragment_padding_size_xsmall">10dp</dimen>
+    <dimen name="dialog_fragment_layout_height_xxbig">200dp</dimen>
```
Comment (nicolbidone) [inline app/src/main/res/values/dimens.xml]: these resources are for all dialog fragments?

File: app/src/main/java/com/onboarding/parkingsystemjava/mvp/view/ConfigureParkingView.java [hunk 23]
```diff
@@ -0,0 +1,45 @@
...
+import android.content.Context;
+import android.widget.Toast;
+import com.onboarding.parkingsystemjava.R;
+import com.onboarding.parkingsystemjava.databinding.ConfigureParkingLotsDialogFragmentMainActivityBinding;
+import com.onboarding.parkingsystemjava.fragment.ConfigureParkingLotsDialog;
+import com.onboarding.parkingsystemjava.mvp.contract.SetParkingContract;
+import java.lang.ref.WeakReference;
+
+public class ConfigureParkingView implements SetParkingContract.SetParkingView {
+    private final WeakReference<ConfigureParkingLotsDialog> fragmentRef;
```
Comment (SartiMau) [inline app/src/main/java/com/onboarding/parkingsystemjava/mvp/view/ConfigureParkingView.java]: please put the weakReference into a FragmentView (just like ActivityView) in the base package

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/onboarding/parkingsystemjava/fragment/ConfigureParkingLotsDialog.java [hunk 24]
```diff
@@ -0,0 +1,62 @@
...
+    private OnInputListener inputListener;
+
+    public static ConfigureParkingLotsDialog newInstance() {
+        return new ConfigureParkingLotsDialog();
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        this.presenter = new ConfigureParkingPresenter(this.inputListener, new ConfigureParkingView(this, binding));
```
Comment (nicolbidone) [inline app/src/main/java/com/onboarding/parkingsystemjava/fragment/ConfigureParkingLotsDialog.java]: why this is in onStart and no in onCreate or onCreateView?

File: app/src/main/java/com/onboarding/parkingsystemjava/fragment/ConfigureParkingLotsDialog.java [hunk 25]
```diff
@@ -0,0 +1,62 @@
...
+        binding.btnDialogFragmentCancel.setOnClickListener(view -> presenter.OnCancelButtonPress());
+    }
+
+    @Override
+    public void onAttach(Context context) {
+        super.onAttach(context);
+        try {
+            inputListener = (OnInputListener) getActivity();
+        } catch (ClassCastException e) {
+            Log.e(ConstantUtils.CONFIGURE_PARKING_LOTS_DIALOG_TAG, ConstantUtils.CLASS_CAST_EXCEPTION + e.getMessage());
```
Comment (nicolbidone) [inline app/src/main/java/com/onboarding/parkingsystemjava/fragment/ConfigureParkingLotsDialog.java]: these strings can be saved as resources

File: app/src/main/java/com/onboarding/parkingsystemjava/mvp/contract/MainActivityContract.java (hunk 7 above)
Comment (nicolbidone) [reply app/src/main/java/com/onboarding/parkingsystemjava/mvp/contract/MainActivityContract.java]: still misnamed

File: app/src/main/java/com/onboarding/parkingsystemjava/activity/MainActivity.java (hunk 19 above)
Comment (Faka-GLB) [reply app/src/main/java/com/onboarding/parkingsystemjava/activity/MainActivity.java:22]: because ParkingPresenter didn't used it.

File: app/src/main/java/com/onboarding/parkingsystemjava/mvp/contract/SetParkingContract.java (hunk 20 above)
Comment (Faka-GLB) [reply app/src/main/java/com/onboarding/parkingsystemjava/mvp/contract/SetParkingContract.java]: done

**Review by nicolbidone — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/onboarding/parkingsystemjava/mvp/presenter/ConfigureParkingPresenter.java (hunk 21 above)
Comment (Faka-GLB) [reply app/src/main/java/com/onboarding/parkingsystemjava/mvp/presenter/ConfigureParkingPresenter.java:19]: you could told me this before I made the empty string constant

File: app/build.gradle (hunk 18 above)
Comment (Faka-GLB) [reply app/build.gradle]: I can't, one of the methods (can't remember which one) requires to have a minSdkversion set to 26

File: app/src/main/res/values/dimens.xml (hunk 22 above)
Comment (Faka-GLB) [reply app/src/main/res/values/dimens.xml]: yes

File: app/src/main/java/com/onboarding/parkingsystemjava/activity/MainActivity.java (hunk 19 above)
Comment (lucianoeli) [reply app/src/main/java/com/onboarding/parkingsystemjava/activity/MainActivity.java:22]: ok. you mean ParkingView

File: app/src/main/java/com/onboarding/parkingsystemjava/mvp/view/ConfigureParkingView.java (hunk 23 above)
Comment (Faka-GLB) [reply app/src/main/java/com/onboarding/parkingsystemjava/mvp/view/ConfigureParkingView.java]: we've discussed that with lucho and he told me it was unnecessary

File: app/build.gradle (hunk 18 above)
Comment (Faka-GLB) [reply app/build.gradle]: image

File: app/src/main/java/com/onboarding/parkingsystemjava/fragment/ConfigureParkingLotsDialog.java (hunk 24 above)
Comment (Faka-GLB) [reply app/src/main/java/com/onboarding/parkingsystemjava/fragment/ConfigureParkingLotsDialog.java]: should I move it?

File: app/src/main/java/com/onboarding/parkingsystemjava/mvp/presenter/ConfigureParkingPresenter.java (hunk 21 above)
Comment (Faka-GLB) [reply app/src/main/java/com/onboarding/parkingsystemjava/mvp/presenter/ConfigureParkingPresenter.java]: done

File: app/src/main/java/com/onboarding/parkingsystemjava/mvp/view/ConfigureParkingView.java (hunk 23 above)
Comment (SartiMau) [reply app/src/main/java/com/onboarding/parkingsystemjava/mvp/view/ConfigureParkingView.java]: please, do it because if you need to use another fragment in the future you will have duplicated code

File: app/build.gradle (hunk 18 above)
Comment (nicolbidone) [reply app/build.gradle]: Facu, clearly you are doing something wrong, in a project, you can't change the min Sdk because you want

File: app/src/main/res/values/dimens.xml (hunk 22 above)
Comment (nicolbidone) [reply app/src/main/res/values/dimens.xml]: ok, which ones? because you only got one

File: app/src/main/java/com/onboarding/parkingsystemjava/mvp/view/ConfigureParkingView.java (hunk 23 above)
Comment (Faka-GLB) [reply app/src/main/java/com/onboarding/parkingsystemjava/mvp/view/ConfigureParkingView.java]: done!

File: app/build.gradle (hunk 18 above)
Comment (Faka-GLB) [reply app/build.gradle]: fixed

File: app/src/main/res/values/dimens.xml (hunk 22 above)
Comment (Faka-GLB) [reply app/src/main/res/values/dimens.xml]: good point, fixed!

File: app/src/main/java/com/onboarding/parkingsystemjava/fragment/ConfigureParkingLotsDialog.java (hunk 24 above)
Comment (Faka-GLB) [reply app/src/main/java/com/onboarding/parkingsystemjava/fragment/ConfigureParkingLotsDialog.java]: moved to onCreateView()

File: app/src/main/java/com/onboarding/parkingsystemjava/fragment/ConfigureParkingLotsDialog.java (hunk 25 above)
Comment (Faka-GLB) [reply app/src/main/java/com/onboarding/parkingsystemjava/fragment/ConfigureParkingLotsDialog.java]: moved to ConstantUtils class

**Review by lucianoeli — APPROVED** [review]

**Review by SartiMau — APPROVED** [review]

**Review by nicolbidone — APPROVED** [review]
